ananas/src/main/java/iamutkarshtiwari/github/io/ananas/editimage/fragment/BaseEditFragment.java

```diff
@@ -1,23 +1,27 @@
 package iamutkarshtiwari.github.io.ananas.editimage.fragment;
 
-import android.os.Bundle;
+import android.content.Context;
+
+import androidx.annotation.NonNull;
 import androidx.fragment.app.Fragment;
 
+import org.jetbrains.annotations.NotNull;
+
 import iamutkarshtiwari.github.io.ananas.editimage.EditImageActivity;
 
 public abstract class BaseEditFragment extends Fragment {
+
     protected EditImageActivity activity;
 
-    protected EditImageActivity ensureEditActivity(){
-        if(activity == null){
+    private void ensureEditActivity() {
```

Comment:
make this protected and get rid of the field maybe?
it's causing a leak

Reply:
Done. Also updated all the affected places with not null checks

Reply:
As talked in chat, we should hold a local variable and not calling the new method all the time